Rubberduck.Core/UI/UnitTesting/TestExplorerViewModel.cs

```diff
@@ -102,49 +89,17 @@ public class TestExplorerViewModel : ViewModelBase, INavigateSelection, IDisposa
             });
         }
 
-        private void RunCompleted(object sender, TestRunEventArgs e)
-        {
-            TotalDuration = e.Duration;
-        }
-
-        private static readonly ParserState[] AllowedRunStates = { ParserState.ResolvedDeclarations, ParserState.ResolvingReferences, ParserState.Ready };
-
-        private bool CanExecuteRunPassedTestsCommand(object obj)
-        {
-            return _vbe.IsInDesignMode && AllowedRunStates.Contains(_state.Status) && Model.Tests.Any(test => test.Result.Outcome == TestOutcome.Succeeded);
-        }
-
-        private bool CanExecuteRunFailedTestsCommand(object obj)
-        {
-            return _vbe.IsInDesignMode && AllowedRunStates.Contains(_state.Status) && Model.Tests.Any(test => test.Result.Outcome == TestOutcome.Failed);
-        }
-
-        private bool CanExecuteRunNotExecutedTestsCommand(object obj)
-        {
-            return _vbe.IsInDesignMode && AllowedRunStates.Contains(_state.Status) && Model.Tests.Any(test => test.Result.Outcome == TestOutcome.Unknown);
-        }
-
-        private bool CanExecuteRunInconclusiveTestsCommand(object obj)
-        {
-            return _vbe.IsInDesignMode && AllowedRunStates.Contains(_state.Status) & Model.Tests.Any(test => test.Result.Outcome == TestOutcome.Inconclusive);
-        }
-
-        private bool CanExecuteRepeatLastRunCommand(object obj)
+        public event EventHandler<TestCompletedEventArgs> TestCompleted;
+        private void TestEngineTestCompleted(object sender, TestCompletedEventArgs e)
         {
-            return _vbe.IsInDesignMode && AllowedRunStates.Contains(_state.Status) && Model.LastRun.Any();
-        }
-
-        public event EventHandler<EventArgs> TestCompleted;
-        private void TestEngineTestCompleted(object sender, EventArgs e)
-        {
-            var handler = TestCompleted;
-            handler?.Invoke(sender, e);
+            // Propagate the event
+            TestCompleted?.Invoke(sender, e);
         }
 
         public INavigateSource SelectedItem => SelectedTest;
 
-        private TestMethod _selectedTest;
-        public TestMethod SelectedTest
+        private TestMethodViewModel _selectedTest;
+        internal TestMethodViewModel SelectedTest
```

Comment:
why is the `SeletedTest` internal, but the rest of the VM state `public`?

Reply:
Because TestMethodViewModel is marked as internal, which means it must not be a public attribute on a public class. I didn't want to move everything to internal, though, so it's basically an artifact of not wanting to force everything into internal for this PR